veps/vep-1.md

```diff
@@ -0,0 +1,90 @@
+
+### Example 1: Deprecating an unsafe of architecturally complex feature
+
+Version `3.xx` of Vitess supports a query that is determined to be dangerous, and/or requires complex architectural support which Vitess needs to be deprecated to improve the code quality of the project:
+
+```
+# A hypothetical dangerous query!
+SELECT 1;
+```
+
+Vitess `4.xx` **MUST** still support this query, because blocking it would be backwards incompatible. However, Vitess `4.xx` **MAY** add an option to _prevent unsafe queries_. It is expected that the default behavior of _prevent unsafe queries_ would be `FALSE`. Executing the query **SHOULD** result in a deprecation warning.
+
+When Vitess `5.xx` is released, the default for _prevent unsafe_queries_ **MAY** be changed to `TRUE`. Waiting until Vitess `5.xx` to change the default makes the behavior _forwards compatible_ because users can test out the change prior to upgrading. Attempting to execute the query **SHOULD** produce an error stating that the `4.xx` behavior can be restored by changing a setting, but the setting is deprecated and will be removed soon.
+
+When Vitess `6.xx` is released, the option to _prevent unsafe queries_ **MAY** be removed with the implied behavior that the option is permanently `TRUE`.
+
+_Based on the impact of functionality being removed, the Vitess team **MAY** at their discretion extend the deprecation cycle. This example serves to demonstrate the minimum time required._
+
+### Example 2: Deprecating and removing an obsolete utility
+
+Version `3.xx` of Vitess supports a utility that has been made obsolete by recently added functionality:
+
+```
+vtdescribe: an obsolete utility
+```
+
+Vitess `4.xx` of Vitess **SHOULD** invoke a warning when attempting to use this utility, with the utility completely removed in Vitess `5.xx`.
+
+_This change is backward compatible in `Vitess 4.xx` because the utility can still be used. It is forwards compatible because attempting to use it will suggest the alternative functionality which is available._
+
+## Supported Fixes to Stable Releases
+
+Because the release includes a new release every 12 weeks, bug fixes will typically only be pushed to the master branch. An exception to this process will be made for high severity bugs, in which case a fix will be available in the weekly release for all currently supported versions.
+
+High severity bugs in this context include CVEs, data corruption, wrong results or outage inducing issues. The Vitess team **MAY** only backport bugs with higher visibility or user impact.
+
+## Supported Upgrade Paths
+
+Because the _Backwards and Forwards Compatibility Promise_ is only between single major versions, it is not recommended to skip versions when upgrading. i.e.
+
+To use _Example 1: Deprecating an unsafe of architecturally complex feature_, a user upgrading from Vitess 3.x to 6.x would encounter a situation where a query works without warning or error, to the query is completely removed.
```

Comment:
```suggestion
To use _Example 1: Deprecating an unsafe or architecturally complex feature_, a user upgrading from Vitess 3.xx to 6.xx would encounter a situation where a query works without warning or error, to the query is completely removed.
```

Reply:
This doesn't read quite right. You might want to read it again and edit as needed.

Reply:
> We should add a section on:
> 
> * `Release Cadence` which explicitly lists the weekly, monthly and quarterly releases.
> * `Release Lifecycle` which explicitly talks about the support term along with what is a backport candidate
> * `Release Artifacts` which explicitly qualifies what will be provided officially

I will try and incorporate the first two (although what is a candidate for support is covered under _Support Fixes to Stable releases_).

If we list release artifacts we imply platforms. It would be nice if platforms policy discussion a little bit detached since platforms should change more frequently than the release cycle.

Reply:
I have updated, PTAL again. Thx :-)